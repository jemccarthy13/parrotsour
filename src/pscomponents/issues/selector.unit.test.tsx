import React from "react"
import { act, fireEvent, render, waitFor } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import IssueSelector from "./selector"

const mockChangeFn = jest.fn(() => {
  return jest.fn()
})

describe("IssueSelector_Component", () => {
  afterEach(() => {
    mockChangeFn.mockClear()
  })

  it("renders_picprob_default_checked", () => {
    const selWrapper = render(<IssueSelector onChange={mockChangeFn} />)

    expect(
      (selWrapper.getByTestId(/iss-pic-selector/i) as HTMLInputElement).checked
    ).toEqual(true)
    expect(selWrapper).toMatchSnapshot()
  })

  it("alerts_parent_only_when_sel_changes", async () => {
    userEvent.setup()
    const selWrapper = render(<IssueSelector onChange={mockChangeFn} />)

    selWrapper.getByTestId(/oth-selector/i).focus()

    act(() => {
      fireEvent.click(selWrapper.getByTestId(/oth-selector/i))
    })

    await waitFor(() => {
      expect(
        (selWrapper.getByTestId(/iss-pic-selector/i) as HTMLInputElement)
          .checked
      ).toEqual(false)
    })

    expect(mockChangeFn).toHaveBeenCalledTimes(1)
    expect(mockChangeFn).toHaveBeenCalledWith("othprob")

    act(() => {
      fireEvent.click(selWrapper.getByTestId(/iss-feature-selector/i))
    })

    await waitFor(() => {
      expect(
        (selWrapper.getByTestId(/iss-oth-selector/i) as HTMLInputElement)
          .checked
      ).toEqual(false)
    })

    expect(mockChangeFn).toHaveBeenCalledTimes(2)
    expect(mockChangeFn).toHaveBeenCalledWith("feature")
  })
})
