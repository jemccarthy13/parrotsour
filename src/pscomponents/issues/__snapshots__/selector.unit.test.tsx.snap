// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSelector_Component renders_picprob_default_checked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pscontainer"
      >
        <h2>
          <u>
            Problem:
          </u>
        </h2>
        <ul>
          <li>
            <input
              checked=""
              data-testid="iss-pic-selector"
              id="picprob"
              name="issue"
              style="color: grey;"
              type="radio"
              value="picprob"
            />
            <label
              for="picprob"
              id="piclbl"
              style="color: grey;"
            >
              The answer to this picture is incorrect
            </label>
            <div
              class="check-no-hover"
            />
          </li>
          <li>
            <input
              data-testid="iss-feature-selector"
              id="feature"
              name="issue"
              style="color: grey;"
              type="radio"
              value="feature"
            />
            <label
              for="feature"
              style="color: grey;"
            >
              Wouldn‘t it be nice if...
            </label>
            <div
              class="check-no-hover"
            />
          </li>
          <li>
            <input
              data-testid="iss-oth-selector"
              id="othprob"
              name="issue"
              type="radio"
              value="othprob"
            />
            <label
              for="othprob"
              style="color: grey;"
            >
              Something else
            </label>
            <div
              class="check-no-hover"
            />
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pscontainer"
    >
      <h2>
        <u>
          Problem:
        </u>
      </h2>
      <ul>
        <li>
          <input
            checked=""
            data-testid="iss-pic-selector"
            id="picprob"
            name="issue"
            style="color: grey;"
            type="radio"
            value="picprob"
          />
          <label
            for="picprob"
            id="piclbl"
            style="color: grey;"
          >
            The answer to this picture is incorrect
          </label>
          <div
            class="check-no-hover"
          />
        </li>
        <li>
          <input
            data-testid="iss-feature-selector"
            id="feature"
            name="issue"
            style="color: grey;"
            type="radio"
            value="feature"
          />
          <label
            for="feature"
            style="color: grey;"
          >
            Wouldn‘t it be nice if...
          </label>
          <div
            class="check-no-hover"
          />
        </li>
        <li>
          <input
            data-testid="iss-oth-selector"
            id="othprob"
            name="issue"
            type="radio"
            value="othprob"
          />
          <label
            for="othprob"
            style="color: grey;"
          >
            Something else
          </label>
          <div
            class="check-no-hover"
          />
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
